### ISSUE TITLE
Add scalafix project validation to CI process

### ISSUE THREAD

**Issue by swoogles** [zio/zio#6040]
Currently, nothing compiles/executes our `scalafix` code.
We should probably make it a submodule in our root `build.sbt`, and execute `scalafmt/test` as one of our CI stages.

**Comment by adamgfraser** [comment]
The disadvantage of this is that it requires any contributor making a change to the ZIO 2.0 API to make the corresponding changes to the Scalafix rule, which may be a somewhat different knowledge base and potentially create a barrier to new contributions.

**Comment by swoogles** [comment]
If we can set things up such that the only command that's executed is `scalafix/test`, then we shouldn't have that problem. It only confirms that transformations we've defined are correctly applied to source code; it's unaffected by broken code in the output directory. 

I know this because we've got all sorts of Stream breakages in there now, but the command still succeeds 😄

If the structure of the scalafix project is too unusual to be a proper submodule in the main project, we could also fall back to just executing that one SBT command as a shell script, since it returns a success code, despite the compilation errors in those other directories.
